### PR TITLE
Update vim plugins

### PR DESCRIPTION
**Pull request by garbas**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by garbas** [comment]
@GrahamcOfBorg build vimPlugins

File: pkgs/misc/vim-plugins/default.nix [hunk 1]
```diff
@@ -148,6 +148,31 @@ rec {
...
+    name = "LanguageClient-neovim-2018-03-06";
+    src = LanguageClient-neovim-src;
+
+    dependencies = [];
+    propogatedBuildInputs = [ LanguageClient-neovim-bin ];
+
+    preFixup = ''
+      substituteInPlace "$out"/share/vim-plugins/LanguageClient-neovim/plugin/LanguageClient.vim \
+        --replace "let l:command = [s:root . '/bin/languageclient']" "let l:command = ['${LanguageClient-neovim-bin}/bin/languageclient']"
+    '';
```
Comment (Mic92) [inline pkgs/misc/vim-plugins/default.nix]: This change was not possible with `pkgs/misc/vim-plugins/vim2nix/additional-nix-code`?

Comment (garbas) [reply pkgs/misc/vim-plugins/default.nix]: @Mic92: @johnchildren tried to do it, but failed ([here](https://github.com/NixOS/nixpkgs/issues/33391#issuecomment-372066712)).

Comment (Mic92) [reply pkgs/misc/vim-plugins/default.nix]: @garbas ok. let's keep it this way. vimPlugins should get a refactoring anyway. It is pretty hacky.

Comment (Mic92) [reply pkgs/misc/vim-plugins/default.nix]: Please also port it to `18.03`.

Comment (garbas) [reply pkgs/misc/vim-plugins/default.nix]: :+1: i agree about the hacky way and the need to refractor it.

**Comment by garbas** [comment]
porting to `release-18.03` branch is happening [here](https://github.com/NixOS/nixpkgs/pull/36860)

**Pull request merged**